### PR TITLE
add a fixed size byte buffer pool for fastproxy 

### PR DESCRIPTION
**Pull request by balinor2017**

### CODE REVIEW

**Comment by balinor2017** [comment]
https://github.com/haxii/fastproxy/issues/37

**Comment by balinor2017** [comment]
add fixed-size byte buffer pool [https://github.com/haxii/fastproxy/issues/37](url)

File: bytebufferpool/fixedsizebytebuffer.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
+package bytebufferpool
+
+import (
+	"io"
+)
+
+// MaxSize default max size for byte buffer
+var MaxSize = 1024 * 1024
```
Comment (cszichao) [inline bytebufferpool/fixedsizebytebuffer.go]: 32 kB default is enough.

File: bytebufferpool/fixedsizebytebuffer.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+// FixedSizeByteBuffer provides fixed size byte buffer
+//
+// ByteBuffer may be used with functions appending data to the given []byte
+// slice.
+//
+// Use Get for obtaining an empty byte buffer.
+type FixedSizeByteBuffer struct {
+	// B is a byte buffer to use in append-like workloads.
+	B    []byte
+	Used int
```
Comment (cszichao) [inline bytebufferpool/fixedsizebytebuffer.go]: `Used`  shouldn't be public, which could be accidentally modified by user, add a Len() func for `FixedSizeByteBuffer ` instead

File: proxy/http/http.go [hunk 3]
```diff
@@ -187,6 +187,9 @@ type Response struct {
 
 	//body http body parser
 	body http.Body
+
+	//pool sniffer bytes buffer pool
+	pool sync.Pool
```
Comment (cszichao) [inline proxy/http/http.go]: what is the `pool` used for?!

**Comment by cszichao** [comment]
where is the corresponding test files?

**Comment by cszichao** [comment]
merge to develop branch instead

**Pull request closed without merging**